DOC-1849: Improve documentation of issue object and a11ychecker_filter_issue option (#2555)
tinymce/tinymce-docs: modules/ROOT/partials/configuration/a11ychecker_filter_issue.adoc

The `+a11ychecker_filter_issue+` option allows Accessibility Checker issues to be removed from the Accessibility Checker report using a callback function. This option is a flexible alternative to the `+a11ychecker_ignored_rules+` option. This option can remove issues from the results shown in the dialog and the `+getReport+` API.

> The function will be passed the results of the xref:a11ychecker.adoc#getreport[`+getReport()+` API]. To filter an issue from the Accessibility Checker report, the callback needs to return `+false+` for that particular issue. The function makes use of xref:a11ychecker.adoc#issue[`issue` objects] as the parameter for filtering issues; after they are generated by the audit, and before being displayed in the dialog.

Reviewer: After clarifying "the results of the report api" I don't think the bit you added is necessary.

It's not clear to me whether this filter is applied during the `getReport` API function call or if integrators need to do that themselves when using the API. I didn't specify that, because I don't have time to check the code, but I think we should.

```suggestion
The callback function will be passed xref:a11ychecker.adoc#issue[`issue` objects] generated by the xref:a11ychecker.adoc#getreport[`+getReport()+` API]. To remove an issue from the Accessibility Checker report the callback needs to return `+false+` for that particular issue.
```

Author: Done, thanks

  plugins: 'a11ychecker',
  toolbar: 'a11ycheck',
  a11ychecker_filter_issue: (issue) => {
	  return !(issue.element.nodeName.toLowerCase() === 'table' || issue.severity !== 'error');
  }
});
----

=== Example: Using `+a11ychecker_filter_issue+` to filter images with empty alternative text from the Accessibility Checker I1 rule

> The callback function in the following example will only return `false` for any issues with `'I1'` as the `+'id'+` image elements with an empty `+'alt+'` attribute, otherwise the issue won't be filtered out. This implementation can be useful as allowing images to have empty alternative text can be an alternative to applying the `+role="presentation"+` attribute to mark an image as `+decorative+`.

Reviewer 2: Is it a bit confusing to use `alternative` in terms of "another option" and `alternative` as in the name of the text in the same sentence?

Author: Good catch, I'll fix that.

> The callback function in the following example will only return `false` for any issues with `'I1'` as the `+'id'+` image elements with an empty `+'alt+'` attribute, otherwise the issue won't be filtered out. This implementation can be useful as allowing images to have empty alternative text can be another method of applying the `+role="presentation"+` attribute to mark an image as `+decorative+`.

Reviewer 3: ```suggestion
The callback function in the following example will only return `false` for any issues with `'I1'` as the `+'id'+` image elements with an empty `+'alt+'` attribute, otherwise the issue won’t be filtered out. This implementation can be useful as allowing images to have empty alternative text can be another method of applying the `+role="presentation"+` attribute to mark an image as `+decorative+`.
```
